File: core/src/main/java/com/bing/lan/core/base/domain/SystemDictionaryQueryObject.java
package com.bing.lan.core.base.domain;

import org.springframework.util.StringUtils;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class SystemDictionaryQueryObject extends QueryObject {
	private String keyword;
	private Long parentId;

	public String getKeyword() {
		return StringUtils.hasLength(keyword) ? keyword : null;
	}
}

File: core/src/main/kotlin/com/bing/lan/core/base/domain/SystemDictionaryQueryObject.kt
package com.bing.lan.core.base.domain

import com.bing.lan.core.base.query.QueryObject
import lombok.Getter
import lombok.Setter
import org.apache.ibatis.type.Alias
import org.springframework.util.StringUtils

@Getter
@Setter
@Alias("SystemDictionaryQueryObject")
class SystemDictionaryQueryObject : QueryObject() {
    private val keyword: String? = null
    val parentId: Long? = null

    fun getKeyword(): String? {
        return if (StringUtils.hasLength(keyword)) keyword else null
    }
}
